kwanghwi/myapp/app.js

```diff
@@ -37,20 +37,156 @@ app.get("/ping", (req, res) => {
 });
 
 app.post("/user/signup", async (req, res) => {
-  const { userId, password, email, profileImage } = req.body;
+  const { userId, password, name, email, profileImage } = req.body;
   await appDataSource.query(
     `INSERT INTO users(
       user_id,
       password,
+      name,
       email,
-      profileImage
-    ) VALUES (?, ?, ?, ?);
+      profile_image
+    ) VALUES (?, ?, ?, ?, ?);
     `,
-    [userId, password, email, profileImage]
+    [userId, password, name, email, profileImage]
   );
   return res.status(201).json({ message: "signup success!" });
 });
 
+app.post("/posts", async (req, res) => {
+  const { title, content, contentImage, userId } = req.body;
+  await appDataSource.query(
+    `INSERT INTO posts(
+      title,
+      content,
+      content_image,
+      user_id
+    ) VALUES (?, ?, ?, ?);
+    `,
+    [title, content, contentImage, userId]
+  );
+  return res.status(201).json({ message: "postcreate success!" });
+});
+
+app.get("/posts", async (req, res) => {
+  await appDataSource.query(
+    `SELECT
+            u.id as user_id,
+            u.profile_image as userProfileImage,
+            p.id as postingId,
+            p.content_image as postingImageUrl,
+            p.content as postingContent
+        FROM posts p
+    INNER JOIN users u ON u.id = p.user_id;
+    `,
+    (err, rows) => {
+      return res.status(200).json({ data: rows });
+    }
+  );
+});
+
+app.get("/users/:userId/posts", async (req, res) => {
```

Comment:
해당 api는 path parameter 에서 userId로 받은 특정 유저가 작성한 글 목록을 불러오는 기능을 하는 것 같네요.
그렇다면 엔드포인트를 지금과 같이 `/posts/:userId` 보다 `/users/:userId/posts` 로 작성하는 것이 조금 더 자명한 엔드포인트를 만들 수 있을 것 같습니다.

아래의 예시 코드를 보면 조금 더 해당 리뷰의 내용이 와닿을 것 같네요-!
이렇듯 자명한 엔드포인트를 작성하는 것에 대해 REST하게 엔드포인트를 작성한다고 합니다.
해당 내용은 이후에 있을 RESTful API 세션에서 다뤄질 예정이니 참고해주세요 :)

```suggestion
// callback 방식의 비동기처리
app.get("/posts", (req, res) => {
    myDataSource.query(`SELECT * FROM users`,
      // 비동기 처리 이후에 실행되는 callback
      (err, data) => {res.status(200).json({ data: data })}
    );
});

// async와 await 방식의 비동기처리
app.get("/posts", async(req, res) => {
  const users = await myDataSource.query(`SELECT * FROM users`);
  
  res.status(200).json({ data : users})
});
```